pkg/cluster/cluster.go

```diff
@@ -0,0 +1,67 @@
+package cluster
```

Comment:
The current appraoch doesn't sound right to me :( 

I'm envisioning something more like, roughtly:

```go
type ClusterInfo struct {
	OperatorNamespace  string
	DistributionImages map[string]string
}

func NewClusterInfo(ctx context.Context, client client.Client) (*ClusterInfo, error) {
  ...
  ...
  do what you have to do
  ...
  ...
  &ClusterInfo{OperatorNamespace: foo, DistributionImages: bar}
}
```

This is easier to test, and we we can embed it later one anywhere like so:

```
type LlamaStackDistributionReconciler struct {
    ClusterInfo *ClusterInfo
}
```

We can even have methods like `IsInitialized()`